# Introduction
Many things have changed since Michael Aldrich invented the earliest form of e-commerce in 1979. Due to COVID-19 lockdowns introduced by governments in 2020, many customers began online shopping and are turning now into loyal online shoppers. 

To make the right decisions about your products and marketing strategies, it is fundamental to take steps to learn and obtain a better understanding of your customers, old and new.

In this dataset hosted by [Kaggle](https://www.kaggle.com/datasets/ytgangster/online-sales-in-usa), we will explore and will use analytical tools to identify purchase patterns and to predict future sales so that we can focus our marketing efforts. 

## Exploratory Data Analysis
EDA is used to understand the underlying structure of the dataset, possible relationships or correlations that hold between covariates, distributions that may affect the election of an algorithm and etcetera.

### Import Data
The data collected is about the online sales of different products, several merchandise and electronic in different in USA. 

Large retailers are actively searching for ways to increase their profit. Sales analysis is one such key techniques used by large retailers to to increase sales by understanding the customers' purchasing behavior & patterns. Market basket analysis examines collections of items to find relationships between items that go together within the business context.

In [10]:
# Importing the libraries

from __future__ import print_function, division
import numpy as np
import pandas as pd

In [12]:
# Loading datasets
onSales_df = pd.read_csv(
    './data/sales.csv'
)

In [13]:
print('{0} observations and {1} characteristics'.format( onSales_df.shape[0], onSales_df.shape[1]))
print('First 5 rows:')
pd.set_option('display.max_columns', None)
onSales_df.head()

286392 observations and 36 characteristics
First 5 rows:


order_id  order_date    status   item_id                  sku  \
0  100354678  2020-10-01  received  574772.0   oasis_Oasis-064-36   
1  100354678  2020-10-01  received  574774.0      Fantastic_FT-48   
2  100354680  2020-10-01  complete  574777.0      mdeal_DMC-610-8   
3  100354680  2020-10-01  complete  574779.0   oasis_Oasis-061-36   
4  100367357  2020-11-13  received  595185.0  MEFNAR59C38B6CA08CD   

   qty_ordered  price   value  discount_amount   total       category  \
0         21.0   89.9  1798.0              0.0  1798.0  Men's Fashion   
1         11.0   19.0   190.0              0.0   190.0  Men's Fashion   
2          9.0  149.9  1199.2              0.0  1199.2  Men's Fashion   
3          9.0   79.9   639.2              0.0   639.2  Men's Fashion   
4          2.0   99.9    99.9              0.0    99.9  Men's Fashion   

  payment_method  bi_st  cust_id  year     month  ref_num Name Prefix  \
0            cod  Valid  60124.0  2020  Oct-2020   987867        Drs.   
1            cod  Valid  60124.0  2020  Oct-2020   987867        Drs.   
2            cod    Net  60124.0  2020  Oct-2020   987867        Drs.   
3            cod    Net  60124.0  2020  Oct-2020   987867        Drs.   
4            cod  Valid  60124.0  2020  Nov-2020   987867        Drs.   

  First Name Middle Initial Last Name Gender   age    full_name  \
0       Jani              W     Titus      F  43.0  Titus, Jani   
1       Jani              W     Titus      F  43.0  Titus, Jani   
2       Jani              W     Titus      F  43.0  Titus, Jani   
3       Jani              W     Titus      F  43.0  Titus, Jani   
4       Jani              W     Titus      F  43.0  Titus, Jani   

                 E Mail Customer Since          SSN    Phone No.  Place Name  \
0  jani.titus@gmail.com      8/22/2006  627-31-5251  405-959-1129     Vinson   
1  jani.titus@gmail.com      8/22/2006  627-31-5251  405-959-1129     Vinson   
2  jani.titus@gmail.com      8/22/2006  627-31-5251  405-959-1129     Vinson   
3  jani.titus@gmail.com      8/22/2006  627-31-5251  405-959-1129     Vinson   
4  jani.titus@gmail.com      8/22/2006  627-31-5251  405-959-1129     Vinson   

   County    City State    Zip Region User Name  Discount_Percent  
0  Harmon  Vinson    OK  73571  South   jwtitus               0.0  
1  Harmon  Vinson    OK  73571  South   jwtitus               0.0  
2  Harmon  Vinson    OK  73571  South   jwtitus               0.0  
3  Harmon  Vinson    OK  73571  South   jwtitus               0.0  
4  Harmon  Vinson    OK  73571  South   jwtitus               0.0

In [14]:
onSales_df.tail()

order_id  order_date      status   item_id                    sku  \
286387  100562365  2021-09-30        paid  905179.0    APPCHA5AF14939B8F8A   
286388  100562376  2021-09-30         cod  905191.0    MEFCOT5A8D1E973B886   
286389  100562383  2021-09-30         cod  905200.0  WOFVAL59D5EA84167F9-M   
286390  100562384  2021-09-30         cod  905202.0  WOFNIG5B4D7EB0E9FDD-L   
286391  100562386  2021-09-30  processing  905205.0    MATHUA5AF70A7D1E50A   

        qty_ordered   price   value  discount_amount   total  \
286387          2.0  4419.9  4419.9              0.0  4419.9   
286388          2.0    39.9    39.9              0.0    39.9   
286389          2.0    40.0    40.0              0.0    40.0   
286390          2.0    49.9    49.9              0.0    49.9   
286391          2.0  3559.9  3559.9              0.0  3559.9   

                 category payment_method  bi_st   cust_id  year     month  \
286387         Appliances        Easypay  Valid  115323.0  2021  Sep-2021   
286388      Men's Fashion            cod  Valid  115324.0  2021  Sep-2021   
286389    Women's Fashion            cod  Valid  115325.0  2021  Sep-2021   
286390    Women's Fashion            cod  Valid  115325.0  2021  Sep-2021   
286391  Mobiles & Tablets    bankalfalah  Gross  115326.0  2021  Sep-2021   

        ref_num Name Prefix First Name Middle Initial  Last Name Gender   age  \
286387   967309       Prof.      Brady              K     Latham      M  51.0   
286388   335358       Prof.     Bennie              M   Brunetti      M  52.0   
286389   675384        Mrs.  Francesca              N     Giusti      F  38.0   
286390   675384        Mrs.  Francesca              N     Giusti      F  38.0   
286391   489455         Mr.       Rolf              E  Schlosser      M  28.0   

                full_name                           E Mail Customer Since  \
286387      Latham, Brady           brady.latham@gmail.com      3/21/2007   
286388   Brunetti, Bennie        bennie.brunetti@gmail.com     10/24/2011   
286389  Giusti, Francesca  francesca.giusti@btinternet.com      7/25/1987   
286390  Giusti, Francesca  francesca.giusti@btinternet.com      7/25/1987   
286391    Schlosser, Rolf       rolf.schlosser@comcast.net      1/28/2015   

                SSN    Phone No.      Place Name    County           City  \
286387  613-87-0361  212-772-7404      Rushville     Yates      Rushville   
286388  101-02-1040  229-817-9451  Lawrenceville  Gwinnett  Lawrenceville   
286389  399-31-7238  252-414-8396         Durham    Durham         Durham   
286390  399-31-7238  252-414-8396         Durham    Durham         Durham   
286391  320-11-8748  423-276-2699      Knoxville      Knox      Knoxville   

       State    Zip     Region    User Name  Discount_Percent  
286387    NY  14544  Northeast     bklatham               0.0  
286388    GA  30044      South   bmbrunetti               0.0  
286389    NC  27701      South     fngiusti               0.0  
286390    NC  27701      South     fngiusti               0.0  
286391    TN  37920      South  reschlosser               0.0

In [15]:
onSales_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 286392 entries, 0 to 286391
Data columns (total 36 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   order_id          286392 non-null  object 
 1   order_date        286392 non-null  object 
 2   status            286392 non-null  object 
 3   item_id           286392 non-null  float64
 4   sku               286392 non-null  object 
 5   qty_ordered       286392 non-null  float64
 6   price             286392 non-null  float64
 7   value             286392 non-null  float64
 8   discount_amount   286392 non-null  float64
 9   total             286392 non-null  float64
 10  category          286392 non-null  object 
 11  payment_method    286392 non-null  object 
 12  bi_st             286392 non-null  object 
 13  cust_id           286392 non-null  float64
 14  year              286392 non-null  int64  
 15  month             286392 non-null  object 
 16  ref_num           28

By observing the data, we quickly notice that there are 11 numeric variables (9 float and 3 integers), 24 objects and 286392. There are none missing or non-available values (Non-null counts in all features is equal to the number of rows 286392). However, there are issues with some columns content e.g., age is not a whole number, same item_id, customer_id, qty_ordered are decimal numbers which make no sense, value doesn't add up with price times qty_ordered and etcetera. 